Midichlorians/Dev/test.cc

```diff
@@ -0,0 +1,129 @@
+SOFTWARE.
+*******************************************************************************/
+
+// Use the define below to enable debug output via `NSLog`.
+// #define CURRENT_APPLE_ENABLE_NSLOG
+
+#include "../../port.h"
+
+#ifdef CURRENT_APPLE
+
+#include "Beta/iOS/Midichlorians.mm"
+#include "Beta/iOS/MidichloriansImpl.mm"
+
+#include "../../Blocks/HTTP/api.h"
+
+#include "../../Bricks/strings/join.h"
+#include "../../Bricks/template/rtti_dynamic_call.h"
+
+#include "../../Bricks/dflags/dflags.h"
+#include "../../3rdparty/gtest/gtest-main-with-dflags.h"
+
+DEFINE_int32(http_port, 8383, "Port to spawn server on.");
+DEFINE_string(http_route, "/log", "HTTP route of the server.");
+
+struct Server {
+  using T_EVENTS = TypeList<iOSAppLaunchEvent, iOSIdentifyEvent, iOSFocusEvent, iOSGenericEvent>;
+
+  Server(int http_port, const std::string& http_route)
+      : messages_processed(0u),
+        endpoints(HTTP(http_port).Register(http_route,
+                                           [this](Request r) {
+                                             std::unique_ptr<MidichloriansEvent> event;
+                                             try {
+                                               CerealizeParseJSON(r.body, event);
+                                               current::metaprogramming::RTTIDynamicCall<T_EVENTS>(event,
+                                                                                                   *this);
+                                               ++messages_processed;
+                                             } catch (const current::Exception&) {
+                                             }
+                                           })) {}
+
+  void operator()(const iOSAppLaunchEvent& event) {
+    EXPECT_FALSE(event.device_id.empty());
+    EXPECT_FALSE(event.binary_version.empty());
+    EXPECT_GT(event.app_install_time, 1420000000000u);
+    EXPECT_GT(event.app_update_time, 1420000000000u);
+    EXPECT_EQ("*FinishLaunchingWithOptions", event.description);
+    messages.push_back("Launch");
+  }
+
+  void operator()(const iOSIdentifyEvent& event) { messages.push_back("Identify[" + event.client_id + ']'); }
+
+  void operator()(const iOSFocusEvent& event) {
+    if (event.gained_focus) {
+      messages.push_back("GainedFocus[" + event.description + ']');
+    } else {
+      messages.push_back("LostFocus[" + event.description + ']');
+    }
+  }
+
+  void operator()(const iOSGenericEvent& event) {
+    std::string params = "source=" + event.source;
+    for (const auto& f : event.fields) {
+      params += ',';
+      params += f.first + '=' + f.second;
+    }
+    for (const auto& f : event.complex_fields) {
+      params += ',';
+      params += f.first + '=' + f.second;
+    }
+    messages.push_back(event.event + '[' + params + ']');
+  }
+
+  std::vector<std::string> messages;
+  std::atomic_size_t messages_processed;
+
+ private:
```

Comment:
Okay, by this point `Server` does look like a class, not struct. The `class_member_` notation in `private:` is in order.